Userland/Libraries/LibWeb/Bindings/OptionConstructor.cpp

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2022, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibWeb/Bindings/HTMLOptionElementPrototype.h>
+#include <LibWeb/Bindings/HTMLOptionElementWrapper.h>
+#include <LibWeb/Bindings/NodeWrapperFactory.h>
+#include <LibWeb/Bindings/OptionConstructor.h>
+#include <LibWeb/DOM/ElementFactory.h>
+#include <LibWeb/HTML/HTMLOptionElement.h>
+#include <LibWeb/HTML/Window.h>
+#include <LibWeb/Namespace.h>
+
+namespace Web::Bindings {
+
+OptionConstructor::OptionConstructor(JS::GlobalObject& global_object)
+    : NativeFunction(*global_object.function_prototype())
+{
+}
+
+void OptionConstructor::initialize(JS::GlobalObject& global_object)
+{
+    auto& vm = this->vm();
+    auto& window = static_cast<WindowObject&>(global_object);
+    NativeFunction::initialize(global_object);
+
+    define_direct_property(vm.names.prototype, &window.ensure_web_prototype<HTMLOptionElementPrototype>("HTMLOptionElement"), 0);
+    define_direct_property(vm.names.length, JS::Value(0), JS::Attribute::Configurable);
+}
+
+JS::ThrowCompletionOr<JS::Value> OptionConstructor::call()
+{
+    return vm().throw_completion<JS::TypeError>(global_object(), JS::ErrorType::ConstructorWithoutNew, "Option");
+}
+
+// https://html.spec.whatwg.org/multipage/form-elements.html#dom-option
+JS::ThrowCompletionOr<JS::Object*> OptionConstructor::construct(FunctionObject&)
+{
+    // 1. Let document be the current global object's associated Document.
+    auto& window = static_cast<WindowObject&>(global_object());
```

Comment:
This isn't the current global object, simply the allocated object's one

Reply:
Oh... I thought this was right because my "reference material" was using it:

https://github.com/SerenityOS/serenity/blob/5bb0b6ba7ab1bcb6e11f081a3103bcd44d597090/Userland/Libraries/LibWeb/Bindings/AudioConstructor.cpp#L41

https://github.com/SerenityOS/serenity/blob/5bb0b6ba7ab1bcb6e11f081a3103bcd44d597090/Userland/Libraries/LibWeb/Bindings/ImageConstructor.cpp#L40

But I guess I got it right now, I will also do another PR to fix the AudioConstructor and ImageConstructor 👍